### ISSUE TITLE
As Open Data Hub manager I would like to migrate the current v1.4 version of OTP to a new v2 version

### ISSUE THREAD

**Issue by rcavaliere** [noi-techpark/odh-mentor-otp#168]

**Comment by rcavaliere** [comment]
Decision: latest version of OTP will be considered

**Comment by leonardehrenfried** [comment]
I estimate the I will take 4 person days for the first draft.

**Comment by leonardehrenfried** [comment]
@dulvui @rcavaliere The GTFS-RT feed returns a 503 error. Is this to be expected?

**Comment by rcavaliere** [comment]
@leonardehrenfried no, it shouldn't be. We already noticed this, are solving the issue with the data provider sharing this feed.

**Comment by leonardehrenfried** [comment]
I checked out the STA feed and it works well with OTP2, however the Skyalps one is invalid. This is (mainly) because the file names called `Agency.txt` rather than `agency.txt`. A quick check also reveals some other errors, like the timezone being `EuropeRome` rather than `Europe/Rome`. There are probably others.

Before this feed can be used, it would need to be made compliant with the GTFS spec.

I recommend that you use the [Mecatran GTFS Vtor](https://gtfsvtor.mecatran.com/utw-test/web/pub/gtfsvtor) to validate your feeds before publishing.

**Comment by leonardehrenfried** [comment]
This is what the OTP2 Debug UI looks like with the STA feed:



![Image](https://github.com/noi-techpark/odh-mentor-otp/assets/151346/39ce0572-c571-4ef0-965f-ca7af74bdcb0)

**Comment by rcavaliere** [comment]
> I estimate the I will take 4 person days for the first draft.

Good, in the meantime let's assign this time to this activity

**Comment by dulvui** [comment]
@leonardehrenfried the flightdata GTFS data has now been refactored and should be valid now.
The report still shows errors, but only for "Too fast travel between stop". We think this error happens because the validator doesn't recognize flight speeds as valid.
Let me know if there are still improvements to do.

**Comment by rcavaliere** [comment]
Status on 9.2: the PR with the OTPv2 instance is available (https://github.com/noi-techpark/odh-mentor-otp/pull/178#issuecomment-1926823763), @dulvui will integrate and evaluate this. Test setup instance has to be prepared.

**Comment by rcavaliere** [comment]
Missing activity to close this issue: automatic script for building OTP with fresh GTFS data - @dulvui

**Comment by dulvui** [comment]
@rcavaliere I simplified now the ci/cd deployment to one nightly build that runs at 3am or trough manual trigger and needs around 20 minutes to build the graph, the docker image and to deploy everything. Now the project is structured like all our other projects and it is much easier to maintain and to understand how it works.

What's still missing here is the gtfs-rt endpoint, that still gives 503 error, so the public transport routes don't show up yet. This is currently affecting also the production deployment on https://mobility.meran.eu with the old otp v1.4

**Comment by rcavaliere** [comment]
@dulvui many thanks! I am currently discussing the GTFS-RT issue with STA,I will let you know.

**Comment by rcavaliere** [comment]
@dulvui wait, it seems that on the current application there is no visibility of GTFS data anymore, can you check?

**Comment by leonardehrenfried** [comment]
I believe it's the same as this: https://github.com/noi-techpark/odh-mentor-otp/pull/178#issuecomment-1953555978

**Comment by dulvui** [comment]
@rcavaliere the production otp works now again. There was a problem with the test and production docker image having the same tags, so somehow the production instance used the new testing otp.
Now test and prod have different and unique tags, so this won't happen again.
